app/scripts/models/Workspace.js

```diff
@@ -204,43 +210,42 @@ define(['backbone', 'Nodes', 'Connection', 'Connections', 'scheme', 'FLOOD', 'Ru
       }, this);
     },
 
-    subscribeOnNodesConnectionsChanges: function(){
+    subscribeOnNodesConnectionsChanges: function() {
 
-      this.get('connections').on('add remove', function () {
-        this.trigger('change:connections');
-        this.trigger('requestRun');
-      }.bind(this));
+        this.listenTo(this.get('connections'), 'add remove', function () {
+            this.trigger('change:connections');
+            this.trigger('requestRun');
+        });
 
-      this.get('nodes').on('add remove', function () {
-        this.trigger('change:nodes');
-        this.trigger('requestRun');
-      }.bind(this));
+        this.listenTo(this.get('nodes'), 'add remove', function () {
+            this.trigger('change:nodes');
+            this.trigger('requestRun');
+        });
     },
 
     customNode : null,
 
-    initializeCustomNode: function(){
+    initializeCustomNode: function() {
 
-      this.customNode = new FLOOD.internalNodeTypes.CustomNode( this.get('name'), this.get('_id'), this.get('guid') );
+        this.customNode = new FLOOD.internalNodeTypes.CustomNode(this.get('name'), this.get('_id'), this.get('guid'));
 
-      var ni = this.get('nodes').where({typeName: "Input"}).length;
-      var no = this.get('nodes').where({typeName: "Output"}).length;
+        var ni = this.get('nodes').where({typeName: "Input"}).length;
+        var no = this.get('nodes').where({typeName: "Output"}).length;
 
-      this.customNode.setNumInputs(ni);
-      this.customNode.setNumOutputs(no);
+        this.customNode.setNumInputs(ni);
+        this.customNode.setNumOutputs(no);
 
-      this.customNode.searchTags = [this.get('name').toLowerCase()];
+        this.customNode.searchTags = [this.get('name').toLowerCase()];
 
-      this.app.SearchElements.addCustomNode( this.customNode );
+        this.app.SearchElements.addCustomNode(this.customNode);
 
-      var that = this;
-
-      this.on('change:name', function(){
-        that.customNode.functionName = that.get('name');
-        that.customNode.searchTags = [that.get('name').toLowerCase()];
-        that.app.SearchElements.addCustomNode( that.customNode );
-      }, this);
+        var that = this;
 
```

Comment:
We could remove this and just use this inside the function body below, as this will get bound correctly.